### PR TITLE
Allow skipping certs validation of Elastic

### PR DESCRIPTION
**Pull request by zarelit**
Co-Authored-By: Carmine Di Monaco <carmine.dimonaco@computer.org>

##### ISSUE TYPE
 - Feature Pull Request

##### SUMMARY
Allow user to skip certificate validation when connecting to the
Elasticsearch cluster (opt-in). It permits to use botkube with clusters
that don't have certificates signed by a CA that is trusted by the
botkube image, it often happens with clusters that are for internal use
only.

See #485

### CODE REVIEW

**Comment by zarelit** [comment]
@PrasadG193 @kartik-moolya what do you think about the PR? Could you please enable running the workflow to allow the build to start?

**Review by kartik-moolya — APPROVED** [review]

**Comment by kartik-moolya** [comment]
@zarelit  Looks good so far, could you 
* Please update the https://github.com/infracloudio/botkube/blob/develop/deploy-all-in-one.yaml and https://github.com/infracloudio/botkube/blob/develop/deploy-all-in-one-tls.yaml with the default values for the new variable.
* We would need to update the helm chart template [here](https://github.com/infracloudio/botkube/tree/develop/helm/botkube) accordingly to accommodate this capability. Please also paste a `--dry run` output after the changes.

Also do you have any logs to show if the ES client was created with the new config?

**Review by kartik-moolya — CHANGES_REQUESTED** [review]
Please refer the comment section for changes/ enahncements requested

**Comment by CDimonaco** [comment]
Hi @kartik-moolya We pushed a new commit with the requested changes, regarding the elastic search logs, currently the application does not log when something is persisted on elastic search. We tested this feature in a test environment with our fork and works well.

File: helm/botkube/README.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ serviceAccount:
 | communications.elasticsearch.password | string | `"ELASTICSEARCH_PASSWORD"` |  |
 | communications.elasticsearch.server | string | `"ELASTICSEARCH_ADDRESS"` |  |
 | communications.elasticsearch.username | string | `"ELASTICSEARCH_USERNAME"` |  |
+| communications.elasticsearch.skipTLSVerify | boolean | `false` | Skip tls verification of elastic search node |
```
Comment (kartik-moolya) [inline helm/botkube/README.md:48]: Let's make it `bool`  that's how it would be everywhere else in the Readme :)

Comment (CDimonaco) [reply helm/botkube/README.md]: Done :D

**Review by kartik-moolya — APPROVED** [review]

File: deploy-all-in-one-tls.yaml [hunk 2]
```diff
@@ -275,6 +275,7 @@ stringData:
         server: 'ELASTICSEARCH_ADDRESS'             # e.g https://example.com:9243
         username: 'ELASTICSEARCH_USERNAME'          # Basic Auth
         password: 'ELASTICSEARCH_PASSWORD'
+        skipTLSVerify: false                      # toggle verification of TLS certificate of the Elastic nodes. Verification is skipped when option is true. Enable to connect to clusters with self-signed certs
```
Comment (PrasadG193) [inline deploy-all-in-one-tls.yaml:278]:
```suggestion
        skipTLSVerify: false                        # toggle verification of TLS certificate of the Elastic nodes. Verification is skipped when option is true. Enable to connect to clusters with self-signed certs
```

File: deploy-all-in-one.yaml [hunk 3]
```diff
@@ -275,6 +275,7 @@ stringData:
         server: 'ELASTICSEARCH_ADDRESS'             # e.g https://example.com:9243
         username: 'ELASTICSEARCH_USERNAME'          # Basic Auth
         password: 'ELASTICSEARCH_PASSWORD'
+        skipTLSVerify: false                      # toggle verification of TLS certificate of the Elastic nodes. Verification is skipped when option is true. Enable to connect to clusters with self-signed certs
```
Comment (PrasadG193) [inline deploy-all-in-one.yaml:278]:
```suggestion
        skipTLSVerify: false                        # toggle verification of TLS certificate of the Elastic nodes. Verification is skipped when option is true. Enable to connect to clusters with self-signed certs
```

File: helm/botkube/values.yaml [hunk 4]
```diff
@@ -321,6 +321,7 @@ communications:
     server: 'ELASTICSEARCH_ADDRESS'             # e.g https://example.com:9243
     username: 'ELASTICSEARCH_USERNAME'          # Basic Auth
     password: 'ELASTICSEARCH_PASSWORD'
+    skipTLSVerify: false                      # toggle verification of TLS certificate of the Elastic nodes. Verification is skipped when option is true. Enable to connect to clusters with self-signed certs
```
Comment (PrasadG193) [inline helm/botkube/values.yaml:324]:
```suggestion
    skipTLSVerify: false                        # toggle verification of TLS certificate of the Elastic nodes. Verification is skipped when option is true. Enable to connect to clusters with self-signed certs
```

**Review by PrasadG193 — APPROVED** [review]
[nit] let's keep uniform indentation for comments

**Comment by zarelit** [comment]
@PrasadG193 @kartik-moolya  is there anything else we can do to unblock the PR journey?

**Comment by PrasadG193** [comment]
Hey @zarelit , looks like the CI is failing after https://github.com/infracloudio/botkube/pull/497. Let me fix that to unblock merging

**Comment by zarelit** [comment]
@PrasadG193  thank you for taking care of the last blocker and the merge

**Pull request merged**